src/chainstate/burn/operations/delegate_stx.rs

```diff
@@ -0,0 +1,290 @@
+                tx.num_signers(),
+                outputs.len()
+            );
+            return Err(op_error::InvalidInput);
+        }
+
+        if outputs.len() == 0 {
+            warn!(
+                "Invalid tx: inputs: {}, outputs: {}",
+                tx.num_signers(),
+                outputs.len()
+            );
+            return Err(op_error::InvalidInput);
+        }
+
+        if tx.opcode() != Opcodes::DelegateStx as u8 {
+            warn!("Invalid tx: invalid opcode {}", tx.opcode());
+            return Err(op_error::InvalidInput);
+        };
+
+        let data = DelegateStxOp::parse_data(&tx.data()).ok_or_else(|| {
+            warn!("Invalid tx data");
+            op_error::ParseError
+        })?;
+
+        let output = outputs[0]
+            .address
+            .clone()
+            .try_into_stacks_address()
+            .ok_or_else(|| {
+                warn!("Invalid tx: output 1 must be representable as a StacksAddress");
+                op_error::InvalidInput
+            })?;
+
+        // coerce a hash mode for this address if need be, since we'll need it when we feed this
+        // address into the .pox contract
+        let reward_addr = if outputs.len() >= 2 {
+            Some(outputs[1].address.clone().coerce_hash_mode())
+        } else {
+            None
+        }; 
+        
+        Ok(DelegateStxOp {
+            sender: sender.clone(),
+            reward_addr: reward_addr,
+            delegate_to: output, 
+            delegated_ustx: data.delegated_ustx,
+            until_burn_height: data.until_burn_height,
+            txid: tx.txid(),
+            vtxindex: tx.vtxindex(),
+            block_height,
+            burn_header_hash: block_hash.clone(),
+        })
+    }
+
+    
+    pub fn check(&self) -> Result<(), op_error> {
+        if self.delegated_ustx == 0 {
+            warn!("Invalid DelegateStxOp, must have positive ustx");
+            return Err(op_error::DelegateStxMustBePositive);
+        }
+
+        Ok(())
+    }
+       
+}
+
+impl StacksMessageCodec for DelegateStxOp {
+    /*
+            Wire format:
+
+        0      2  3                     24               40 41       50
+        |------|--|---------------------|----------------|--|--------|
+         magic  op  delegate to           delegated ustx  opt       until burn height
+                                                          
+
+    */
+    fn consensus_serialize<W: Write>(&self, fd: &mut W) -> Result<(), codec_error> {
+        write_next(fd, &(Opcodes::DelegateStx as u8))?;
+        self.delegate_to.consensus_serialize(fd)?;
+        fd.write_all(&self.delegated_ustx.to_be_bytes())
+            .map_err(|e| codec_error::WriteError(e))?;
+        write_next(fd, &(self.reward_addr.is_some() as u8))?;
+        write_next(fd, &(self.until_burn_height.is_some() as u8))?;
+
+        // Q-JUDE
+        if let Some(height) = self.until_burn_height {
```

Comment:
I think you're missing an `fd.write_all(&(1 as u8))` here to mark the presence of an 8-byte height to follow.